src/app/AttributeCache.cpp

```diff
@@ -46,6 +52,33 @@ CHIP_ERROR AttributeCache::UpdateCache(const ConcreteDataAttributePath & aPath,
         handle.RightSize();
 
         state.Set<System::PacketBufferHandle>(std::move(handle));
+
+        //
+        // Clear out the committed data version and only set it again once we have received all data for this cluster.
+        // Otherwise, we may have incomplete data that looks like it's complete since it has a valid data version.
+        //
+        mCache[aPath.mEndpointId][aPath.mClusterId].mCommittedDataVersion.ClearValue();
+
+        for (const auto & path : mRequestPathSet)
+        {
+            if (!path.IncludesAllAttributesInCluster(aPath))
```

Comment:
This is odd. The logic to update the committed version for the 'last' attribute path to the pending version should not be predicated on the new/current path being wildcard.

Reply:
updated, thanks